core/src/main/java/org/apache/druid/java/util/common/parsers/JSONFlattenerMaker.java

```diff
@@ -53,16 +53,23 @@
                    .options(EnumSet.of(Option.SUPPRESS_EXCEPTIONS))
                    .build();
 
+  private final boolean keepNullValues;
+
   private final CharsetEncoder enc = StandardCharsets.UTF_8.newEncoder();
 
+  public JSONFlattenerMaker(boolean keepNullValues)
+  {
+    this.keepNullValues = keepNullValues;
+  }
+
   @Override
   public Iterable<String> discoverRootFields(final JsonNode obj)
   {
     return FluentIterable.from(() -> obj.fields())
                          .filter(
                              entry -> {
                                final JsonNode val = entry.getValue();
-                               return !(val.isObject() || val.isNull() || (val.isArray() && !isFlatList(val)));
+                               return !(val.isObject() || (!keepNullValues && val.isNull()) || (val.isArray() && !isFlatList(val)));
```

Comment:
I'm wondering why `JSONFlattenMaker` filters nulls out even though other `FlattenMaker` implementations don't. We should probably make the behavior consistent, but it should be done in a separate PR.

Reply:
This comment is not a blocker for this PR, but more like a note.